### PR TITLE
Expose TypeInfo constructor

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/corefx/issues/14334

Any other changes that are involved and should be done as part of this PR?

### CODE REVIEW

**Comment by ViktorHofer** [comment]
@dotnet/dnceng finished leg doesn't report back to Github that it's completed: https://ci.dot.net/job/dotnet_corefx/job/master/job/CROSS_check_prtest/16195/. It still says pending. I will merge this now.

**Comment by ViktorHofer** [comment]
thanks

File: src/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -6279,7 +6279,7 @@ public partial class TypeDelegator : System.Reflection.TypeInfo
     public delegate bool TypeFilter(System.Type m, object filterCriteria);
     public abstract partial class TypeInfo : System.Type, System.Reflection.IReflectableType
     {
-        internal TypeInfo() { }
+        protected TypeInfo() { }
```
Comment (weshaggard) [inline src/System.Runtime/ref/System.Runtime.cs]: Is this already protected in coreclr and uap implementation?

Comment (weshaggard) [reply src/System.Runtime/ref/System.Runtime.cs]: Looks like it is https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/Reflection/TypeInfo.cs#L11. 
https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Reflection/TypeInfo.cs#L11

**Comment by weshaggard** [comment]
👍

**Pull request merged**